src/internal/group_api/mod.rs

```diff
@@ -359,88 +372,153 @@ pub(crate) fn get_group_keys<'a>(
-                                    &auth.signing_private_key().into(),
-                                );
-                                maybe_transform_key.map(|transform_key| {
-                                    (id, (user_pub_key, Some(transform_key.into())))
-                                })
+    use futures::future::Either;
+    let owner_id = match owner.clone() {
+        Some(id) => id,
+        None => auth.account_id().clone(),
+    };
+    // don't make any requests if validation fails
+    if admins.is_empty() {
+        Either::A(err(IronOxideErr::ValidationError(
+            format!("admins"),
+            format!("admins list cannot be empty"),
+        )))
+    } else if !admins.contains(&owner_id) {
+        Either::A(err(IronOxideErr::ValidationError(
+            format!("admins"),
+            format!("admins list must contain the owner"),
+        )))
+    } else {
+        Either::B(
+            user_api::user_key_list(auth, users_to_lookup)
+                .and_then(move |user_ids_and_keys| {
+                    // this will occur when one of the UserIds cannot be found
+                    if user_ids_and_keys.len() != users_to_lookup.len() {
+                        // figure out which user ids could not be found in the database and include them in the error message.
+                        let desired_users_set: HashSet<UserId> =
+                            HashSet::from_iter(users_to_lookup.clone());
+                        let found_users_set: HashSet<UserId> =
+                            user_ids_and_keys.into_iter().map(|(x, _)| x).collect();
+                        let diff: HashSet<&UserId> =
+                            desired_users_set.difference(&found_users_set).collect();
+                        err(IronOxideErr::UserDoesNotExist(format!(
+                            "Failed to find the following users: {:?}",
+                            diff
+                        )))
+                    } else {
+                        transform::gen_group_keys(recrypt)
+                            .and_then(move |(plaintext, group_priv_key, group_pub_key)| {
+                                let (member_info_result, admin_info): (
+                                    Result<Vec<(UserId, PublicKey, TransformKey)>, IronOxideErr>,
+                                    Vec<(UserId, PublicKey)>,
+                                ) = {
+                                    let members_set: HashSet<_> = HashSet::from_iter(&members);
+                                    let admins_set: HashSet<_> = HashSet::from_iter(&admins);
+                                    let mut member_info: Vec<
+                                        Result<(UserId, PublicKey, TransformKey), IronOxideErr>,
+                                    > = vec![];
+                                    let mut admin_info: Vec<(UserId, PublicKey)> = vec![];
+                                    user_ids_and_keys
```

Comment:
Would it be easier to put the ids back into two lists earlier: members and admins -- to simplify this? 
Step 1: partition the user_ids_and_keys
Step 2: Generate transform keys for members

Reply:
I wasn't meaning literal `partition`, but the idea of separating the members and admins earlier. I still think this might be a good idea, but I'll look at the latest code and see...

Reply:
You implemented what I was thinking, except for the combining of the transform key generation.